src/components/tl/dpu/tl_dpu_context.c

```diff
@@ -0,0 +1,246 @@
+    ucp_ep_params_t     ep_params;
+    ucp_ep_h            ucp_ep;
+    ucp_context_h       ucp_context;
+    ucp_worker_h        ucp_worker;
+    ucc_status_t        status = UCC_OK;
+    int ret;
+
+    /* Identify DPU */
+    char hname[256];
+    void *rem_worker_addr;
+    size_t rem_worker_addr_size;
+    int sockfd = 0, dpu_found = 0;
+
+    UCC_CLASS_CALL_SUPER_INIT(ucc_tl_context_t, tl_dpu_config->super.tl_lib,
+                              params->context);
+    memcpy(&self->cfg, tl_dpu_config, sizeof(*tl_dpu_config));
+
+    /* Find  DPU based on the host-dpu list */
+    gethostname(hname, sizeof(hname) - 1);
+    if (tl_dpu_config->use_dpu) {
+        char *h = calloc(1, 256);
+        FILE *fp = NULL;
+
+        if (strcmp(tl_dpu_config->host_dpu_list,"") != 0) {
+
+            fp = fopen(tl_dpu_config->host_dpu_list, "r");
+            if (fp == NULL) {
+                fprintf(stderr, "Unable to open \"%s\", disabling dpu team\n", tl_dpu_config->host_dpu_list);
+                tl_dpu_config->use_dpu = 0;
+                dpu_found = 0;
+            }
+            else {
+                while (fscanf(fp,"%s", h) != EOF) {
+                    if (strcmp(h, hname) == 0) {
+                        dpu_found = 1;
+                        fscanf(fp, "%s", hname);
+                        fprintf(stderr, "DPU <%s> found!\n", hname);
+                        break;
+                    }
+                    memset(h, 0, 256);
+                }
+            }
+            if (!dpu_found) {
+                tl_dpu_config->use_dpu = 0;
+            }
+        }
+        else {
+            fprintf(stderr, "DPU_ENABLE set, but HOST_LIST not specified. Disabling DPU team!\n");
+            tl_dpu_config->use_dpu = 0;
+        }
+        free(h);
+    }
+    else {
+        goto err;
+    }
+
+    if (!dpu_found) {
+        goto err;
+    }
+
+    fprintf(stdout, "Connecting to %s", hname);
+    sockfd = _server_connect(hname, tl_dpu_config->server_port);
+
+    memset(&ucp_params, 0, sizeof(ucp_params));
+    ucp_params.field_mask      = UCP_PARAM_FIELD_FEATURES |
+                                 UCP_PARAM_FIELD_REQUEST_SIZE |
+                                 UCP_PARAM_FIELD_REQUEST_INIT |
+                                 UCP_PARAM_FIELD_REQUEST_CLEANUP;
+    ucp_params.features        = UCP_FEATURE_TAG |
+                                 UCP_FEATURE_RMA;
+    ucp_params.request_size    = sizeof(tl_dpu_ucp_request_t);
+    ucp_params.request_init    = tl_dpu_ucp_req_init;
+    ucp_params.request_cleanup = tl_dpu_ucp_req_cleanup;
+
+    status = ucp_init(&ucp_params, NULL, &ucp_context);
+    if (status != UCC_OK) {
+        fprintf(stderr, "failed ucp_init(%s)\n", ucc_status_string(status));
+        goto err;
+    }
+
+    memset(&worker_params, 0, sizeof(worker_params));
+    worker_params.field_mask    = UCP_WORKER_PARAM_FIELD_THREAD_MODE;
+    worker_params.thread_mode   = UCS_THREAD_MODE_SINGLE;
```

Comment:
Generic question: does TL/DPU supports THREAD_MULTIPLE? I assume thread_mode for ucp worker must be taken from params coming from CORE (similarly to TL/UCP)